src/main/kotlin/uk/gov/justice/hmpps/offenderevents/services/HMPPSDomainEventsEmitter.kt

```diff
@@ -505,6 +507,46 @@ class HMPPSDomainEventsEmitter(
 
       else -> emptyList()
     }
+
+  private fun VideoAppointmentCancelledEvent.Companion.toDomainEvents(event: VideoAppointmentCancelledEvent): List<HmppsDomainEvent> =
+    event.toDomainEvent().toListOrEmptyWhenNull()
+
+  private val videoAppointmentTypes = listOf("VLB", "VLPM", "VLLA", "VLOO", "VLPA", "VLAP")
+
+  private fun VideoAppointmentCancelledEvent.toDomainEvent(): HmppsDomainEvent? =
+    when {
+      // Only raise an event for the 6 video appointment types, and only if they have been cancelled or deleted
+      (this.recordDeleted || this.scheduleEventStatus == "CANC") && videoAppointmentTypes.contains(this.scheduleEventSubType) -> {
+        prisonApiService.getPrisonerNumberForBookingId(this.bookingId).getOrNull()?.let {
```

Comment:
although I guess you're doing it this way since you only need to get the prison number for a cancelled video appointment?

Reply:
Yes - I was debating this - but then considered that there is one existing consumer of the APPOINTMENT_CHANGED event directly from the offender-events topic (whereabouts-api), and altering the message structure to add `offenderIdDisplay` posed a risk to that. I also noticed other events converting the bookingId to nomsId in `hmpps-prisoner-events` and thought it might be safer here. I don't mind either way - this is service in your care, so (taking into account the other direct consumer) just let me know the preference please?

Reply:
It's only needed for the much smaller number of a) Video-type appointments - that's 6/117 types, and b) where they are cancelled (as you mentioned). So any fewer than doing it "at source" for all appointment inserts, updates and deletes.